bigchaindb_driver/pool.py

```diff
@@ -63,7 +75,45 @@ def __init__(self, connections, picker_class=RoundRobinPicker):
 
         """
         self.connections = connections
+        self.node_count = len(self.connections)
+        self.max_retries = self.node_count * 4
```

Comment:
Fix me if I am wrong but it looks like `get_connection` will forever return `None` after the retry maximum is reached..

You can remove the cap and instead add [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) - in the simplest form it would look like:

```
delay = initial_delay * 2 ** retries
```
This will allow the driver users to wait for a successful connection for as long as they wish while optimizing the load put onto the network.

Also, makes sense to start with a small delay, some fraction of a second. The client should not wait for a minute if it's an intermittent network failure.

Reply:
i agree and you were not wrong, please check next push.